src/engine/Events.ts

```diff
@@ -16,7 +16,7 @@ export type prekill = 'prekill';
 export type postkill = 'postkill';
```

Comment:
Also, as You can see, i moved all propagation functionality to event itself, after did some research about this in internet and books and discussing it with my friends. I find it more suitable to address all this responsibilities to event itself. And it  now has more adequate structure.